src/agent/onefuzz-supervisor/src/scheduler.rs

```diff
@@ -129,17 +145,36 @@ impl State<Free> {
 pub enum SetupDone {
     Ready(State<Ready>),
     PendingReboot(State<PendingReboot>),
+    Done(State<Done>),
 }
 
 impl State<SettingUp> {
     pub async fn finish(self, runner: &mut dyn ISetupRunner) -> Result<SetupDone> {
         let work_set = self.ctx.work_set;
 
-        let output = runner.run(&work_set).await?;
-
-        if let Some(output) = output {
-            if !output.exit_status.success {
-                anyhow::bail!("setup script failed: {:?}", output);
+        let output = runner.run(&work_set).await;
+
+        match output {
+            Ok(Some(output)) => {
+                if !output.exit_status.success {
```

Comment:
With this change, if the setup script fails, we transition to `Done`, with context.

Reply:
This branch is for when the setup script ran, but had a nonzero exit.